### ISSUE TITLE
Sort symlinks by their origin

### ISSUE THREAD

**Issue by sixsixfive** [lxde/pcmanfm-qt#347]
a comparision of pcman(right) and thunar(left). All red remove icons point to the same file :

Pcmanfm(both qt and gtk versions) sorts all files by size, thunar does this much better since it will firstly sort them by their link target.

![bildschirmfoto7](https://cloud.githubusercontent.com/assets/3780774/14770471/9de3e200-0a72-11e6-996b-51a0a21b6792.png)

**Comment by tsujan** [comment]
> Pcmanfm sorts all files by size

What do you mean by that? PCManFM sorts files according to what its user chooses: name, modification time or size.

**Comment by sixsixfive** [comment]
it does it by size, however it ignores the link target(if the files are symlinks) eg: look at the pcmanfm screenshot the remove symlinks/icons are mixed with other symlinks that have the same size, thunar on the other side sorts them all togehter which gives a nice overview

**Comment by tsujan** [comment]
Do you mean *if* you choose sorting by size, *then* all symlinks come first? And do you propose, *in the case of sorting by size*, they should be sorted according the sizes of their targets?

**Comment by sixsixfive** [comment]
sorting by size, then all symlinks come first

no.

I mean there should be a second filter applied that displays/places all symlinks(that point to the same file -since they have all the same size) together.

maybe this screenshot in thunars detail mode shows it more:

![bildschirmfoto8](https://cloud.githubusercontent.com/assets/3780774/14771029/1b7d0b04-0a7f-11e6-8ada-a0fb5ffd6bb8.png)

 files are sorted by size, however all files that point to the same target are put together

**Comment by tsujan** [comment]
Sometimes it's hard to get what you mean, even with screenshots ;)

This is my last guess, until you (or someone else) find a clearer way of telling what you mean: If the user chooses sorting by size, then all symlinks should come first but they should be ordered according to the sizes of their targets.

**Comment by palinek** [comment]
I think @sixsixfive wants the size sorting like (pseudocode):
```c++
bool operator <(const File& a, const File& b)
{
  return a.size() < b.size()
      || (a.size() == b.size() && a.isSymlink() && b.isSymlink() && dynamic_cast<const Symlink&>(a).targetFilename() < dynamic_cast<const Symlink&>(a).targetFilename() );
}
```

@tsujan does this make sense to you?

**Comment by tsujan** [comment]
> I think @sixsixfive wants the size sorting like (pseudocode)...

If so, that was why I didn't understand it because it doesn't make sense to me.

However, IMO, https://github.com/lxde/pcmanfm-qt/issues/347#issuecomment-214059199 makes sense. @palinek What do you think?

**Comment by palinek** [comment]
> However, IMO, #347 (comment) makes sense. @palinek What do you think?

All said here is IMO way too specific. And I would say that only a general sorting should be done. What I think (based on the (detail) screenshot) thunar does is "sorting by size (in case of symlink, the size of target is used) and filename"

**Comment by tsujan** [comment]
> All said here is IMO way too specific.

I agree. I should see how PCManFM sorts symlinks. IMO, in the case of sorting by name, their own names should be used and, in the case of sorting by size, the sizes of their targets. The latter will also give @sixsixfive what he wants.

@palinek BTW, for some reason unknown to me, I understand your words better. Would you please tell me whether I've correctly understood what is discussed at https://github.com/lxde/lxqt/issues/1030, especially the last comments (@Vladimir-csp's). I mean, whenever you have time.

**Comment by Vladimir-csp** [comment]
It could be generalized as 'follow symlinks' sorting option. If enabled, target file's stats would be used for sorting (whether it is size, mtime or type), if not - only ink's itself. @sixsixfive, what do you think?

**Comment by sixsixfive** [comment]
> 'follow symlinks' sorting option. 

It would be Ok, but why an extra option for this? I think the best would be to make this the default behaivor(like it is in thunar, caja, nautilus, konqueror & dolphin)
